### PR TITLE
Run3-hcx345 Try to fix ieta=16 issue for HCAL simulation in the GEN-SIM step

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Try to fix ieta=16 issue for HCAL simulation in the GEN-SIM step

#### PR validation:

Tested using the script SimG4CMS/Calo/test/python/ttbar.py

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

May need to backport a part of this PR to CMSSW versions 13_0_X and 13_1_X

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by abdoulline** [comment]
@bsunanda 
test with 1k muon gun shot at |ieta|=16 (eta=1.3-1.4, phi= -pi +pi) 
confirmed that there is **no any** instance of |ieta|=16 depth1 with the weight **other** than 1.0.

**Comment by perrotta** [comment]
@bsunanda @abdoulline could any of you expert summarize in the PR description what is the "ieta=16 issue" that this PR wants to fix?
These PRs are often a bit cryptic if one wants to understand what the problem was and how do you plan to fix it...

File: Geometry/HcalCommonData/src/HcalDDDSimConstants.cc [hunk 1]
```diff
@@ -4,24 +4,24 @@
 #include "FWCore/Utilities/interface/Exception.h"
 #include "DataFormats/Math/interface/GeantUnits.h"
 
-//#define EDM_ML_DEBUG
+#define EDM_ML_DEBUG
```
Comment (civanch) [inline Geometry/HcalCommonData/src/HcalDDDSimConstants.cc:7]: @bsunanda , please, comment out debugging.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
There is layer 1 in eta=16. Sometimes the edge of eta=15 in layer 1 is assigned as eta=16. Though these are rare cases, this happens and it is avoided here. So it is a safer version

File: Geometry/HcalCommonData/src/HcalDDDSimConstants.cc (hunk 1 above)
Comment (bsunanda) [reply Geometry/HcalCommonData/src/HcalDDDSimConstants.cc]: done

**Comment by bsunanda** [comment]
@civanch Please approve it now

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**